src/OpenTelemetry/Internal/SelfDiagnosticsConfigParser.cs

```diff
@@ -0,0 +1,116 @@
+// <copyright file="SelfDiagnosticsConfigParser.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace OpenTelemetry.Internal
+{
+    internal class SelfDiagnosticsConfigParser
+    {
+        private const string ConfigFileName = "DiagnosticsConfiguration.json";
```

Comment:
Wonder if we want a name that is more associated with OpenTelemetry, e.g. `OTEL_DIAG.json`?

Reply:
Makes sense, but I wonder if we could align the naming closer to https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-environment-variables.md#general-sdk-configuration as `OTEL_LOG.json` or having the ["self-diagnostics"](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#self-diagnostics) name in it? @cijothomas

Reply:
Note to myself: update #1477 if this changes.

Reply:
OTEL_Diagnostics_Config.json? 
We definitely should prefix "Otel". Rest is okay. We can even use Otel_Self_Diagnostics
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/error-handling.md#self-diagnostics

Reply:
Changed to `OTEL_DIAGNOSTICS.json` in the PR above. Making all upper case to follow the examples in https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-environment-variables.md